src/sdk/wallet/providers/wallet-connect-2.wallet-provider.ts

```diff
@@ -0,0 +1,60 @@
+import { BytesLike } from 'ethers';
+import { toHex } from '../../common';
+import { DynamicWalletProvider } from './dynamic.wallet-provider';
+import { EthereumProvider } from './interfaces';
+
+export class WalletConnect2WalletProvider extends DynamicWalletProvider {
+  constructor(readonly provider: EthereumProvider) {
+    super('WalletConnect2');
+
+    try {
+      const {
+        accounts: [address],
+        chainId,
+      } = provider;
+
+      this.setAddress(address);
+      this.setNetworkName(chainId);
+    } catch (err) {
+      //
+    }
```

Comment:
Just out of curiosity is there a reason we're using try / catch here as if `accounts` / `chainId` are not found in provider it would just return `undefined` for the destrucutred vartiables?

Reply:
copy paste implementation of previous Wallet Connect provider